dictionaries/software-terms/src/coding-compound-terms.txt

```diff
@@ -0,0 +1,118 @@
+# Coding Related Compound Terms
+# Compound terms are words that are often compouned with outher terms in code.
+# Be careful with compounds, they can be combined in any combination.
+# cspell:ignoreRegExp /#.*/g
+# Add new terms after this line, one per line.
+*auth*
+*auto*
+*balance*
+*balancer*
+*balancing*
+*batch*
+*batched*
+*big*
+*bigquery*
+*bin*
+*box*
+*browser*
+*buf*
+*buffer*
+*buffered*
+*buffering*
+*buffers*
+*button*
+*buttons*
+*channel*
+*clear*
+*client*
+*cloud*
+*code*
+*commit*
+*connect*
+*connection*
+*data*
+*data*
+*data*
+*domain*
+*down*
+*enable*
+*encrypt*
+*error*
+*field*
+*fix*
+*gateway*
+*left*
+*limit*
+*limited*
+*limiter*
+*limiting*
+*live*
+*load*
+*meta*
+*migration*
+*name*
+*net*
+*network*
+*out*
+*over*
+*plane*
+*plot*
+*policy*
+*query*
+*rate*
+*reservation*
+*right*
+*scale*
+*scale*
+*scaled*
+*scaler*
+*scaling*
+*server*
+*service*
+*slice*
+*sliced*
+*source*
+*split*
+*sqrt*
+*square*
+*storage*
+*store*
+*stream*
+*switch*
+*text*
+*transfer*
+*trigger*
+*triggered*
+*up*
+*zip*
++goog
++io
++std
+goog+
+h+ # common prefix like `HKEY_LOCAL_HOST`
+i+ # Only use as a prefix, like `ilimit` or `istream`
+io+
+my*
+o+ # Only use as a prefix, like `oauth` or `ostream`
```

Comment:
I would have not added directly the H I O ones. I would have added the most commonly used words using those prefix, because here it allows a lot of false positive

`inline` is a word, `iline` could be consider as a typo for `inline`, but now it won't be reported anymore

Reply:
Good point.

Another option, is to have everything with leading caps:

`I+` and `*Line*`.

Based upon the `caseSensitive` setting, it can prevent false negatives in normal text:

When writing text and `caseSensitive` is `true`, it will not match against `iline`, only `ILine`. When writing code and `caseSensitive` is `false`, it will match.

Reply:
Addressed in #3937

Reply:
I commented 

https://github.com/streetsidesoftware/cspell-dicts/issues/3857#issuecomment-2563485226

"You will have to document at some point you know" *said mommy by rolling her eyes*